featurebyte/service/online_store_table_version.py

```diff
@@ -0,0 +1,93 @@
+"""
+OnlineStoreTableVersionService class
+"""
+from __future__ import annotations
+
+from typing import Dict, Optional
+
+from featurebyte.exception import DocumentNotFoundError
+from featurebyte.models.online_store_table_version import (
+    OnlineStoreTableVersion,
+    OnlineStoreTableVersionUpdate,
+)
+from featurebyte.service.base_document import BaseDocumentService
+
+
+class OnlineStoreTableVersionService(
+    BaseDocumentService[
+        OnlineStoreTableVersion, OnlineStoreTableVersion, OnlineStoreTableVersionUpdate
+    ]
+):
+    """
+    OnlineStoreTableVersionService class
+
+    This service is used to manage the current active version of every aggregation result names in
+    the online store tables.
+    """
+
+    document_class = OnlineStoreTableVersion
+    document_update_class = OnlineStoreTableVersionUpdate
+
+    async def get_version(self, aggregation_result_name: str) -> Optional[int]:
+        """
+        Get the version of an aggregation result name
+
+        Parameters
+        ----------
+        aggregation_result_name: str
+            Aggregation result name
+
+        Returns
+        -------
+        Optional[int]
+            Version of the OnlineStoreTableVersion
+        """
+        query_filter = {"aggregation_result_name": aggregation_result_name}
+        async for doc in self.list_documents_iterator(query_filter):
+            return int(doc["version"])
+        return None
+
+    async def get_versions(self, aggregation_result_names: list[str]) -> Dict[str, int]:
+        """
+        Get the versions for a list of aggregation result names
+
+        Parameters
+        ----------
+        aggregation_result_names: list[str]
+            List of aggregation result name
+
+        Returns
+        -------
+        Dict[str, int]
+            Versions of the OnlineStoreTableVersion
+        """
+        query_filter = {"aggregation_result_name": {"$in": aggregation_result_names}}
```

Comment:
We should build mongo index on `aggregation_result_name` to avoid collection scan.

Reply:
What would be the procedure of doing that?

Reply:
Add an entry in model settting, similar to https://github.com/featurebyte/featurebyte/blob/d909054884c9e995194d96f2ec3b71d127a88081/featurebyte/models/feature.py#L86.

Reply:
Opps... Sorry that i missed that part that it is introduced in the model.